pkg/host/store.go

```diff
@@ -128,10 +129,16 @@ func (s *Store[E]) Update(_ context.Context, obj E) (E, error) {
 		return obj, nil
 	}
 
+	if oldObj.GetResourceVersion() != obj.GetResourceVersion() {
+		return utils.Zero[E](), fmt.Errorf("failed to update object: resourceVersion is not latest")
```

Comment:
could you please create exporter error which can be used in other parts of code for detect this problem.

```
var ErrResourceVersionNotLatest = errors.New("resourceVersion is not latest")

fmt.Errorf("failed to update object: %w, ErrResourceVersionNotLatest)
```